--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -2255,7 +2255,6 @@ void RenderViewContextMenu::AppendPageIt
                                   IDS_CONTENT_CONTEXT_SAVEPAGEAS);
   menu_model_.AddItemWithStringId(IDC_PRINT, IDS_CONTENT_CONTEXT_PRINT);
   AppendLiveCaptionItem();
-  AppendMediaRouterItem();
   if (IsRegionSearchEnabled()) {
     AppendRegionSearchItem();
   }
@@ -2383,8 +2382,7 @@ void RenderViewContextMenu::AppendMediaR
 
 void RenderViewContextMenu::AppendReadingModeItem() {
   // Show Read Anything option if it's not already open in the side panel.
-  if (GetBrowser() && GetBrowser()->is_type_normal() &&
-      !IsReadAnythingEntryShowing(GetBrowser())) {
+  if ((false)) {
     menu_model_.AddItemWithStringId(IDC_CONTENT_CONTEXT_OPEN_IN_READING_MODE,
                                     IDS_CONTENT_CONTEXT_READING_MODE);
   }
@@ -3635,6 +3633,7 @@ void RenderViewContextMenu::ExecuteComma
 }
 
 void RenderViewContextMenu::AddSpellCheckServiceItem(bool is_checked) {
+  return;
   AddSpellCheckServiceItem(&menu_model_, is_checked);
 }
 
@@ -4005,6 +4004,8 @@ bool RenderViewContextMenu::IsVideoFrame
 bool RenderViewContextMenu::AppendQRCodeGeneratorItem(bool for_image,
                                                       bool draw_icon,
                                                       bool add_separator) {
+  return false;
+
   if (!IsQRCodeGeneratorEnabled()) {
     return false;
   }
--- a/chrome/browser/ui/browser_tab_strip_model_delegate.cc
+++ b/chrome/browser/ui/browser_tab_strip_model_delegate.cc
@@ -304,6 +304,7 @@ void BrowserTabStripModelDelegate::AddTo
 }
 
 bool BrowserTabStripModelDelegate::SupportsReadLater() {
+  return false;
   return !browser_->profile()->IsGuestSession() && !IsForWebApp();
 }
 
