--- a/services/network/public/cpp/features.cc
+++ b/services/network/public/cpp/features.cc
@@ -490,7 +490,7 @@ BASE_FEATURE(kUpdateRequestForCorsRedire
 // Kill switch for the Topics API.
 BASE_FEATURE(kBrowsingTopics,
              "BrowsingTopics",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enable the shared storage API. Note that enabling this feature does not
 // automatically expose this API to the web, it only allows the element to be
@@ -574,7 +574,7 @@ BASE_FEATURE(kSharedStorageTransactional
 // API exposure will be disabled regardless of the OT config.
 BASE_FEATURE(kInterestGroupStorage,
              "InterestGroupStorage",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 // TODO(crbug.com/40176812): Adjust these limits in response to usage.
 BASE_FEATURE_PARAM(int,
                    kInterestGroupStorageMaxOwners,
