--- a/ui/base/pointer/touch_ui_controller.cc
+++ b/ui/base/pointer/touch_ui_controller.cc
@@ -175,13 +175,7 @@ void TouchUiController::TouchUiScoperFor
 // static
 TouchUiController* TouchUiController::Get() {
   static base::NoDestructor<TouchUiController> instance([] {
-    const std::string switch_value =
-        base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
-            switches::kTopChromeTouchUi);
-    if (switch_value == switches::kTopChromeTouchUiDisabled)
-      return TouchUiState::kDisabled;
-    const bool enabled = switch_value == switches::kTopChromeTouchUiEnabled;
-    return enabled ? TouchUiState::kEnabled : TouchUiState::kAuto;
+    return TouchUiState::kDisabled;
   }());
   return instance.get();
 }
