--- a/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc
+++ b/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc
@@ -1279,6 +1279,9 @@ gfx::Rect FedCmAccountSelectionView::Get
 
 void FedCmAccountSelectionView::ShouldShowDialog(bool& should_show) {
   if (dialog_type_ == DialogType::BUBBLE) {
+    should_show = false;
+    return;
+
     // Hide the bubble dialog if it can't fit.
     if (!CanFitInWebContents()) {
       should_show = false;
