--- a/chrome/browser/upgrade_detector/upgrade_detector_impl.cc
+++ b/chrome/browser/upgrade_detector/upgrade_detector_impl.cc
@@ -214,6 +214,8 @@ void UpgradeDetectorImpl::DoCalculateThr
 
 void UpgradeDetectorImpl::StartOutdatedBuildDetector() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  return;
+
   static BASE_FEATURE(kOutdatedBuildDetector, "OutdatedBuildDetector",
                       base::FEATURE_ENABLED_BY_DEFAULT);
 
