--- a/chrome/browser/ui/toolbar/pinned_toolbar/pinned_toolbar_actions_model.cc
+++ b/chrome/browser/ui/toolbar/pinned_toolbar/pinned_toolbar_actions_model.cc
@@ -240,8 +240,7 @@ void PinnedToolbarActionsModel::MaybeMig
     UpdatePinnedState(kActionShowChromeLabs, true);
     pref_service_->SetBoolean(prefs::kPinnedChromeLabsMigrationComplete, true);
   }
-  if (features::HasTabSearchToolbarButton() &&
-      !pref_service_->GetBoolean(prefs::kTabSearchMigrationComplete)) {
+  if ((false)) {
     UpdatePinnedState(kActionTabSearch, true);
     pref_service_->SetBoolean(prefs::kTabSearchMigrationComplete, true);
   }
--- a/chrome/browser/ui/toolbar/toolbar_pref_names.cc
+++ b/chrome/browser/ui/toolbar/toolbar_pref_names.cc
@@ -23,7 +23,7 @@ void RegisterProfilePrefs(user_prefs::Pr
     default_pinned_actions.Append(chrome_labs_action.value());
   }
 
-  if (features::HasTabSearchToolbarButton()) {
+  if ((false)) {
     const std::optional<std::string>& tab_search_action =
         actions::ActionIdMap::ActionIdToString(kActionTabSearch);
     if (tab_search_action.has_value()) {
--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -671,7 +671,7 @@ BASE_FEATURE(kTabStripBrowserApi,
 
 BASE_FEATURE(kTabstripComboButton,
              "TabstripComboButton",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // This serves as a "kill-switch" for migrating the Tab Search feature to be a
 // toolbar button for non-ChromeOS users in the US.
@@ -707,6 +707,8 @@ bool HasTabSearchToolbarButton() {
     return false;
   }
 
+  return true;
+
   static const bool is_tab_search_moving = [] {
     if (GetCountryCode() == "us" &&
         base::FeatureList::IsEnabled(
