--- a/components/download/public/common/download_features.cc
+++ b/components/download/public/common/download_features.cc
@@ -11,11 +11,7 @@ namespace features {
 
 BASE_FEATURE(kParallelDownloading,
              "ParallelDownloading",
-#if BUILDFLAG(IS_ANDROID)
              base::FEATURE_ENABLED_BY_DEFAULT
-#else
-             base::FEATURE_DISABLED_BY_DEFAULT
-#endif
 );
 
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
