--- a/components/performance_manager/features.cc
+++ b/components/performance_manager/features.cc
@@ -248,7 +248,7 @@ BASE_FEATURE(kRecordFreezingEligibilityU
 
 BASE_FEATURE(kInfiniteTabsFreezing,
              "InfiniteTabsFreezing",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE_PARAM(int,
                    kInfiniteTabsFreezing_NumProtectedTabs,
