--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -891,7 +891,7 @@ BASE_FEATURE(HttpsFirstBalancedMode, bas
 // Automatically enables HTTPS-First Mode in a balanced configuration when
 // possible.
 BASE_FEATURE(HttpsFirstBalancedModeAutoEnable,
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Kill switch for crbug.com/1414633.
 BASE_FEATURE(kHttpsFirstModeForAdvancedProtectionUsers,
