--- a/chrome/browser/profiles/profiles_state.cc
+++ b/chrome/browser/profiles/profiles_state.cc
@@ -326,7 +326,7 @@ std::u16string GetDefaultNameForNewEnter
     name = base::UTF8ToUTF16(hosted_domain);
   } else {
     name = l10n_util::GetStringUTF16(
-        IDS_SIGNIN_DICE_WEB_INTERCEPT_ENTERPRISE_PROFILE_NAME);
+        IDS_SINGLE_PROFILE_DISPLAY_NAME);
   }
   CHECK(!name.empty());
   return name;
