--- a/components/omnibox/browser/autocomplete_match.cc
+++ b/components/omnibox/browser/autocomplete_match.cc
@@ -654,7 +654,7 @@ const gfx::VectorIcon& AutocompleteMatch
           case KEYWORD_MODE_STARTER_PACK_HISTORY:
             return vector_icons::kHistoryChromeRefreshIcon;
           case KEYWORD_MODE_STARTER_PACK_TABS:
-            return omnibox::kProductChromeRefreshIcon;
+            return omnibox::kTabIcon;
           case KEYWORD_MODE_STARTER_PACK_GEMINI:
             return omnibox::kSparkIcon;
           case KEYWORD_MODE_STARTER_PACK_AI_MODE:
--- a/components/omnibox/browser/vector_icons/tab.icon
+++ b/components/omnibox/browser/vector_icons/tab.icon
@@ -2,14 +2,21 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-CANVAS_DIMENSIONS, 32,
-MOVE_TO, 3, 26,
-LINE_TO, 7, 12,
-LINE_TO, 24, 12,
-LINE_TO, 28, 26,
-LINE_TO, 24, 26,
-LINE_TO, 22, 16,
-LINE_TO, 9, 16,
-LINE_TO, 7, 26,
-LINE_TO, 3, 26,
+MOVE_TO, 42, 6,
+H_LINE_TO, 6,
+R_CUBIC_TO, -2.21, 0, -4, 1.79, -4, 4,
+R_V_LINE_TO, 28,
+R_CUBIC_TO, 0, 2.21, 1.79, 4, 4, 4,
+R_H_LINE_TO, 36,
+R_CUBIC_TO, 2.21, 0, 4, -1.79, 4, -4,
+V_LINE_TO, 10,
+R_CUBIC_TO, 0, -2.21, -1.79, -4, -4, -4,
+CLOSE,
+R_MOVE_TO, 0, 32,
+H_LINE_TO, 6,
+V_LINE_TO, 10,
+R_H_LINE_TO, 20,
+R_V_LINE_TO, 8,
+R_H_LINE_TO, 16,
+R_V_LINE_TO, 20,
 CLOSE
