--- a/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_frame_toolbar_utils.h
+++ b/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_frame_toolbar_utils.h
@@ -11,7 +11,7 @@ class ToolbarButton;
 class ToolbarButtonProvider;
 
 #if BUILDFLAG(IS_MAC)
-inline constexpr int kWebAppMenuMargin = 7;
+inline constexpr int kWebAppMenuMargin = 4;
 #endif
 
 // Makes adjustments to |toolbar_button| for display in a web app frame.
--- a/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_toolbar_button_container.cc
+++ b/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_toolbar_button_container.cc
@@ -85,7 +85,7 @@ WebAppToolbarButtonContainer::WebAppTool
       .SetDefault(
           views::kMarginsKey,
           gfx::Insets::VH(
-              0, HorizontalPaddingBetweenPageActionsAndAppMenuButtons()))
+              0, 9))
       .SetCollapseMargins(true)
       .SetIgnoreDefaultMainAxisMargins(true)
       .SetCrossAxisAlignment(views::LayoutAlignment::kCenter)
