--- a/chrome/browser/ui/views/frame/immersive_mode_controller_mac.mm
+++ b/chrome/browser/ui/views/frame/immersive_mode_controller_mac.mm
@@ -239,11 +239,9 @@ gfx::Insets ImmersiveModeControllerMac::
     right_left_inset = 66;
   }
 
-  // Without this +1 top inset the tabs sit 1px too high. I assume this is
-  // because in fullscreen there is no resize handle.
   return browser_view_->frame()->GetFrameView()->CaptionButtonsOnLeadingEdge()
-             ? gfx::Insets::TLBR(1, right_left_inset, 0, 0)
-             : gfx::Insets::TLBR(1, 0, 0, right_left_inset);
+             ? gfx::Insets::TLBR(0, right_left_inset, 0, 0)
+             : gfx::Insets::TLBR(0, 0, 0, right_left_inset);
 }
 // LINT.ThenChange(//chrome/browser/ui/views/frame/browser_non_client_frame_view_mac.mm:MacTabStripInsets)
 
