--- a/chrome/browser/ui/views/location_bar/icon_label_bubble_view.cc
+++ b/chrome/browser/ui/views/location_bar/icon_label_bubble_view.cc
@@ -49,7 +49,7 @@ constexpr int kIconLabelBubbleSeparatorW
 
 // Amount of space on either side of the separator that appears after the icon
 // or label.
-constexpr int kIconLabelBubbleSpaceBesideSeparator = 8;
+constexpr int kIconLabelBubbleSpaceBesideSeparator = 6;
 
 // The length of the separator's fade animation. These values are empirical.
 constexpr int kIconLabelBubbleFadeInDurationMs = 250;
@@ -85,9 +85,7 @@ void IconLabelBubbleView::SeparatorView:
   const SkColor separator_color = SkColorSetA(
       owner_->delegate_->GetIconLabelBubbleSurroundingForegroundColor(),
       owner_->delegate_->GetIconLabelBubbleSeparatorAlpha());
-  const float x = GetLocalBounds().right() -
-                  owner_->GetEndPaddingWithSeparator() -
-                  1.0f / canvas->image_scale();
+  const float x = 0.5f;
   canvas->DrawLine(gfx::PointF(x, GetLocalBounds().y()),
                    gfx::PointF(x, GetLocalBounds().bottom()), separator_color);
 }
@@ -286,8 +284,7 @@ views::ProposedLayout IconLabelBubbleVie
   // value to be, since it won't be visible.
   const int label_x = image_bounds_with_label.right() + GetInternalSpacing();
   const int available_label_width =
-      std::max(0, layout.host_size.width() - label_x - bubble_trailing_padding -
-                      GetWidthBetweenIconAndSeparator());
+      std::max(0, layout.host_size.width() - label_x - bubble_trailing_padding);
   gfx::Rect label_bounds(label_x, 0, available_label_width, height);
   layout.child_layouts.emplace_back(
       label(),
@@ -309,10 +306,9 @@ views::ProposedLayout IconLabelBubbleVie
   gfx::Rect separator_bounds(label_bounds);
   separator_bounds.Inset(
       gfx::Insets::VH((separator_bounds.height() - separator_height) / 2, 0));
-  float separator_width =
-      GetWidthBetweenIconAndSeparator() + GetEndPaddingWithSeparator();
+  float separator_width = 2;
   int separator_x =
-      label()->GetText().empty() ? image_bounds.right() : label_bounds.right();
+      (label()->GetText().empty() ? image_bounds.right() : label_bounds.right()) + 2;
 
   layout.child_layouts.emplace_back(
       const_cast<views::View*>(this->separator_view()), ShouldShowSeparator(),
@@ -624,7 +620,9 @@ gfx::RoundedCornersF IconLabelBubbleView
   if (radii_.has_value()) {
     return radii_.value();
   }
-  return gfx::RoundedCornersF(GetPreferredSize().height() / 2);
+
+  // location bar rounding is 8px and padding is 2px, so inner rounding is 6px
+  return gfx::RoundedCornersF(6.0f);
 }
 
 void IconLabelBubbleView::SetCornerRadii(const gfx::RoundedCornersF& radii) {
@@ -672,7 +670,7 @@ int IconLabelBubbleView::GetInternalSpac
   }
 
   constexpr int kDefaultInternalSpacingTouchUI = 10;
-  constexpr int kDefaultInternalSpacingChromeRefresh = 4;
+  constexpr int kDefaultInternalSpacingChromeRefresh = 7;
 
   return (ui::TouchUiController::Get()->touch_ui()
               ? kDefaultInternalSpacingTouchUI
@@ -825,6 +823,10 @@ SkPath IconLabelBubbleView::GetHighlight
       SkIntToScalar(radii.lower_right()), SkIntToScalar(radii.lower_right()),
       SkIntToScalar(radii.lower_left()),  SkIntToScalar(radii.lower_left())};
 
+  const SkRect new_rect = RectToSkRect(highlight_bounds);
+  const int layout_radius = GetLayoutConstant(LOCATION_BAR_CHILD_CORNER_RADIUS);
+  return SkPath().addRoundRect(new_rect, layout_radius, layout_radius);
+
   return SkPath().addRoundRect(rect, sk_radii);
 }
 
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -227,7 +227,10 @@ LocationBarView::LocationBarView(Browser
                  !v->GetOmniboxPopupView()->IsOpen();
         }));
     views::FocusRing::Get(this)->SetOutsetFocusRingDisabled(true);
-    views::InstallPillHighlightPathGenerator(this);
+
+    // same border radius as the location bar itself
+    views::InstallRoundRectHighlightPathGenerator(
+      this, gfx::Insets(), GetBorderRadius());
 
 #if BUILDFLAG(OS_LEVEL_GEOLOCATION_PERMISSION_SUPPORTED)
     if (features::IsOsLevelGeolocationPermissionSupportEnabled()) {
@@ -277,9 +280,9 @@ void LocationBarView::Init() {
       CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_PRIMARY);
 
   const gfx::FontList& omnibox_chip_font_list = typography_provider.GetFont(
-      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_BODY_4_EMPHASIS);
+      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_PRIMARY);
   const gfx::FontList& page_action_font_list = typography_provider.GetFont(
-      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_BODY_3_EMPHASIS);
+      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_PRIMARY);
 
   auto location_icon_view =
       std::make_unique<LocationIconView>(omnibox_chip_font_list, this, this);
@@ -398,21 +401,11 @@ void LocationBarView::Init() {
     // first so that they appear on the left side of the icon container.
     // TODO(crbug.com/40835681): Improve the ordering heuristics for page action
     // icons and determine a way to handle simultaneous icon animations.
-    if (base::FeatureList::IsEnabled(commerce::kProductSpecifications)) {
-      params.types_enabled.push_back(
-          PageActionIconType::kProductSpecifications);
-    }
-    params.types_enabled.push_back(PageActionIconType::kDiscounts);
-    params.types_enabled.push_back(PageActionIconType::kPriceInsights);
-    params.types_enabled.push_back(PageActionIconType::kPriceTracking);
-
     if (optimization_guide::features::ShouldEnableOptimizationGuideIconView()) {
       params.types_enabled.push_back(PageActionIconType::kOptimizationGuide);
     }
     params.types_enabled.push_back(PageActionIconType::kClickToCall);
     params.types_enabled.push_back(PageActionIconType::kSmsRemoteFetcher);
-    params.types_enabled.push_back(PageActionIconType::kAutofillAddress);
-    params.types_enabled.push_back(PageActionIconType::kManagePasswords);
     if (!apps::features::ShouldShowLinkCapturingUX()) {
       params.types_enabled.push_back(PageActionIconType::kIntentPicker);
     }
@@ -422,54 +415,8 @@ void LocationBarView::Init() {
     params.types_enabled.push_back(PageActionIconType::kZoom);
     params.types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
-    params.types_enabled.push_back(PageActionIconType::kCookieControls);
-    params.types_enabled.push_back(
-        PageActionIconType::kPaymentsOfferNotification);
     params.types_enabled.push_back(PageActionIconType::kMemorySaver);
   }
-  params.types_enabled.push_back(PageActionIconType::kSaveCard);
-  params.types_enabled.push_back(PageActionIconType::kSaveIban);
-  params.types_enabled.push_back(PageActionIconType::kFilledCardInformation);
-  params.types_enabled.push_back(PageActionIconType::kVirtualCardEnroll);
-  params.types_enabled.push_back(PageActionIconType::kMandatoryReauth);
-
-  if (browser_ &&
-      base::FeatureList::IsEnabled(omnibox::kAiModeOmniboxEntryPoint)) {
-    // Position in the leading position, like the entrypoint for
-    // kLensOverlayHomework below. While both chips may be enabled, they will
-    // not appear at the same time due to different focus behavior. The
-    // visibility of this entrypoint is dependent on whether or not the user
-    // meets AIM eligibility criteria.
-    params.types_enabled.insert(params.types_enabled.begin(),
-                                PageActionIconType::kAiMode);
-  }
-
-  if (browser_ && lens::features::IsOmniboxEntryPointEnabled()) {
-    // The persistent compact entrypoint should be positioned directly before
-    // the star icon and the prominent expanding entrypoint should be
-    // positioned in the leading position. This entrypoint will be suppressed
-    // if the AIM page action is enabled and the user meets AIM eligibility
-    // criteria, since we want to avoid both showing up when the user focuses
-    // the Omnibox.
-    if (lens::features::IsOmniboxEntrypointAlwaysVisible()) {
-      params.types_enabled.push_back(PageActionIconType::kLensOverlay);
-    } else {
-      params.types_enabled.insert(params.types_enabled.begin(),
-                                  PageActionIconType::kLensOverlay);
-    }
-  }
-
-  if (browser_ && lens::features::IsLensOverlayEduActionChipEnabled()) {
-    // Position in the leading position, like the expanding entrypoint for
-    // kLensOverlay above. While both chips may be enabled, they will not appear
-    // at the same time due to different focus behavior.
-    params.types_enabled.insert(params.types_enabled.begin(),
-                                PageActionIconType::kLensOverlayHomework);
-  }
-
-  if (browser_ && tab_groups::SavedTabGroupUtils::SupportsSharedTabGroups()) {
-    params.types_enabled.push_back(PageActionIconType::kCollaborationMessaging);
-  }
 
   if (browser_ && !is_popup_mode_) {
     params.types_enabled.push_back(PageActionIconType::kBookmarkStar);
@@ -510,7 +457,7 @@ bool LocationBarView::IsInitialized() co
 
 int LocationBarView::GetBorderRadius() const {
   return ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
-      views::Emphasis::kMaximum, size());
+      views::Emphasis::kHigh, size());
 }
 
 std::unique_ptr<views::Background> LocationBarView::CreateRoundRectBackground(
@@ -710,9 +657,9 @@ void LocationBarView::Layout(PassKey) {
   // TODO(manukh): Move constants to layout_constants.cc.
   // The padding between the left edges of the location bar and the LHS icon,
   // e.g. the page info icon, the google G icon, the selected suggestion icon.
-  int icon_left = 5;
+  int icon_left = 2;
   // The padding between the LHS icon and the text.
-  int text_left = 8;
+  int text_left = 5;
 
   // Apply indentation to align the omnibox input icon and the text with those
   // of the suggestions in the popup. However, there are two exceptions where
@@ -731,22 +678,22 @@ void LocationBarView::Layout(PassKey) {
                              !location_icon_view_->ShouldShowLabel() &&
                              !ShouldShowKeywordBubble();
   if (should_indent) {
-    icon_left += 7 /*icon_indent*/;
+    icon_left += 0 /*icon_indent*/;
     icon_left += omnibox_feature_configs::AdjustOmniboxIndent()
                      .Get()
                      .input_icon_indent_offset;
-    text_left += 6 /*text_indent*/;
+    text_left += 0 /*text_indent*/;
     text_left += omnibox_feature_configs::AdjustOmniboxIndent()
                      .Get()
                      .input_text_indent_offset;
   } else if (ShouldShowKeywordBubble()) {
     // Otherwise, if in keyword mode, adjust indentation to align the icon and
     // the text with the suggestion icons & texts.
-    icon_left += 9;  /*icon_indent_keyword_mode*/
+    icon_left += 0;  /*icon_indent_keyword_mode*/
     icon_left += omnibox_feature_configs::AdjustOmniboxIndent()
                      .Get()
                      .input_icon_indent_offset;
-    text_left += -9; /*text_indent_keyword_mode*/
+    text_left += 0; /*text_indent_keyword_mode*/
   }
 
   LocationBarLayout leading_decorations(LocationBarLayout::Position::kLeftEdge,
@@ -1235,7 +1182,7 @@ int LocationBarView::GetMinimumTrailingW
 }
 
 gfx::Rect LocationBarView::GetLocalBoundsWithoutEndcaps() const {
-  const int border_radius = height() / 2;
+  const int border_radius = GetBorderRadius();
   gfx::Rect bounds_without_endcaps(GetLocalBounds());
   bounds_without_endcaps.Inset(gfx::Insets::VH(0, border_radius));
   return bounds_without_endcaps;
@@ -1249,9 +1196,9 @@ void LocationBarView::RefreshBackground(
   const bool high_contrast = GetNativeTheme()->UserHasContrastPreference();
 
   const auto* const color_provider = GetColorProvider();
-  SkColor normal = color_provider->GetColor(kColorLocationBarBackground);
+  SkColor normal = color_provider->GetColor(kColorOmniboxResultsBackground);
   SkColor hovered =
-      color_provider->GetColor(kColorLocationBarBackgroundHovered);
+      color_provider->GetColor(kColorOmniboxResultsBackgroundHovered);
 
   background_color_ = gfx::Tween::ColorValueBetween(opacity, normal, hovered);
   if (is_caret_visible) {
--- a/chrome/browser/ui/views/location_bar/location_icon_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_icon_view.cc
@@ -110,7 +110,7 @@ SkColor LocationIconView::GetForegroundC
 }
 
 bool LocationIconView::ShouldShowSeparator() const {
-  return false;
+  return ShouldShowLabel();
 }
 
 bool LocationIconView::ShouldShowLabelAfterAnimation() const {
@@ -372,7 +372,7 @@ void LocationIconView::Update(bool suppr
   // level.
   UpdateLabelColors();
 
-  if (force_hide_background) {
+  if (true) {
     SetBackgroundColor(SK_ColorTRANSPARENT);
   }
 
@@ -458,7 +458,7 @@ gfx::Size LocationIconView::GetMinimumSi
 }
 
 void LocationIconView::SetBackgroundColor(SkColor color) {
-  SetBackground(views::CreateRoundedRectBackground(color, GetCornerRadii()));
+  SetBackground(views::CreateRoundedRectBackground(SK_ColorTRANSPARENT, GetCornerRadii()));
 }
 
 BEGIN_METADATA(LocationIconView)
