--- a/chrome/browser/ui/views/profiles/profile_menu_view.cc
+++ b/chrome/browser/ui/views/profiles/profile_menu_view.cc
@@ -811,6 +811,7 @@ void ProfileMenuView::MaybeBuildBatchUpl
 }
 
 void ProfileMenuView::BuildAutofillSettingsButton() {
+  return;
   CHECK(!profile().IsGuestSession());
   AddFeatureButton(
       l10n_util::GetStringUTF16(IDS_PROFILE_MENU_AUTOFILL_SETTINGS_BUTTON),
@@ -892,6 +893,7 @@ void ProfileMenuView::MaybeBuildChromeAc
 }
 
 void ProfileMenuView::MaybeBuildManageGoogleAccountButton() {
+  return;
   CHECK(!profile().IsGuestSession());
   signin::IdentityManager* identity_manager =
       IdentityManagerFactory::GetForProfile(&profile());
