--- a/components/password_manager/core/browser/password_suggestion_generator.cc
+++ b/components/password_manager/core/browser/password_suggestion_generator.cc
@@ -96,6 +96,8 @@ Suggestion::PasswordSuggestionDetails Ge
 }
 
 void MaybeAppendManagePasswordsEntry(std::vector<Suggestion>* suggestions) {
+  return;
+
   bool has_no_fillable_suggestions = std::ranges::none_of(
       *suggestions,
       [](SuggestionType id) {
