--- a/chrome/browser/ui/views/frame/tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/tab_strip_region_view.cc
@@ -85,7 +85,7 @@ class FrameGrabHandle : public views::Vi
     // is full.
     // TODO(tbergquist): Define this relative to the NTB insets again.
     if (base::CommandLine::ForCurrentProcess()->HasSwitch("remove-grab-handle")) return gfx::Size(0, 0);
-    return gfx::Size(42, 0);
+    return gfx::Size(29, 0);
   }
 };
 
