--- a/chrome/browser/ui/views/tabs/tab_strip_control_button.cc
+++ b/chrome/browser/ui/views/tabs/tab_strip_control_button.cc
@@ -239,7 +239,7 @@ void TabStripControlButton::UpdateBackgr
 }
 
 int TabStripControlButton::GetCornerRadius() const {
-  return TabStripControlButton::kButtonSize.width() / 2;
+  return 8;
 }
 
 int TabStripControlButton::GetFlatCornerRadius() const {
--- a/chrome/browser/ui/views/tabs/tab_search_button.cc
+++ b/chrome/browser/ui/views/tabs/tab_search_button.cc
@@ -23,7 +23,7 @@
 #include "ui/views/view_class_properties.h"
 
 namespace {
-constexpr int kCRTabSearchCornerRadius = 10;
+constexpr int kCRTabSearchCornerRadius = 8;
 constexpr int kCRTabSearchFlatCornerRadius = 4;
 constexpr int kComboButtonFlatCornerRadius = 0;
 }  // namespace
@@ -44,15 +44,6 @@ TabSearchButton::TabSearchButton(
   SetTooltipText(l10n_util::GetStringUTF16(IDS_TOOLTIP_TAB_SEARCH));
   GetViewAccessibility().SetName(
       l10n_util::GetStringUTF16(IDS_ACCNAME_TAB_SEARCH));
-
-  CHECK(!features::HasTabSearchToolbarButton());
-  SetForegroundFrameActiveColorId(kColorNewTabButtonForegroundFrameActive);
-  SetForegroundFrameInactiveColorId(kColorNewTabButtonForegroundFrameInactive);
-  SetBackgroundFrameActiveColorId(kColorNewTabButtonCRBackgroundFrameActive);
-  SetBackgroundFrameInactiveColorId(
-      kColorNewTabButtonCRBackgroundFrameInactive);
-
-  UpdateColors();
 }
 
 TabSearchButton::~TabSearchButton() = default;
