--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -787,6 +787,7 @@ class BrowserViewLayoutDelegateImpl : pu
   }
 
   bool IsContentsSeparatorEnabled() const override {
+    return false;
     // Web app windows manage their own separator.
     // TODO(crbug.com/40102629): Make PWAs set the visibility of the ToolbarView
     // based on whether it is visible instead of setting the height to 0px. This
--- a/chrome/browser/ui/views/frame/top_container_background.cc
+++ b/chrome/browser/ui/views/frame/top_container_background.cc
@@ -6,6 +6,7 @@
 
 #include "chrome/browser/themes/theme_properties.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
+#include "chrome/browser/ui/views/frame/browser_non_client_frame_view.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 #include "chrome/browser/ui/views/tabs/tab_strip.h"
 #include "chrome/grit/theme_resources.h"
@@ -68,6 +69,7 @@ void TopContainerBackground::PaintBackgr
                                              const BrowserView* browser_view) {
   bool painted = PaintThemeCustomImage(canvas, view, browser_view);
   if (!painted) {
-    canvas->DrawColor(view->GetColorProvider()->GetColor(kColorToolbar));
+    canvas->DrawColor(browser_view->frame()->GetFrameView()->GetFrameColor(
+        BrowserFrameActiveState::kUseCurrent));
   }
 }
--- a/chrome/browser/ui/views/frame/contents_separator.cc
+++ b/chrome/browser/ui/views/frame/contents_separator.cc
@@ -4,20 +4,9 @@
 
 #include "chrome/browser/ui/views/frame/contents_separator.h"
 
-#include "chrome/browser/ui/color/chrome_color_id.h"
 #include "ui/base/metadata/metadata_impl_macros.h"
-#include "ui/views/background.h"
-#include "ui/views/controls/separator.h"
 
-ContentsSeparator::ContentsSeparator() {
-  SetBackground(
-      views::CreateSolidBackground(kColorToolbarContentAreaSeparator));
-
-  // BrowserViewLayout will respect either the height or width of this,
-  // depending on orientation, not simultaneously both.
-  SetPreferredSize(
-      gfx::Size(views::Separator::kThickness, views::Separator::kThickness));
-}
+ContentsSeparator::ContentsSeparator() {}
 
 BEGIN_METADATA(ContentsSeparator)
 END_METADATA
