--- a/net/dns/public/doh_provider_entry.cc
+++ b/net/dns/public/doh_provider_entry.cc
@@ -252,7 +252,7 @@ const DohProviderEntry::List& DohProvide
         LoggingLevel::kNormal},
        {"Quad9Secure",
         MAKE_BASE_FEATURE_WITH_STATIC_STORAGE(
-            DohProviderQuad9Secure, base::FEATURE_DISABLED_BY_DEFAULT),
+            DohProviderQuad9Secure, base::FEATURE_ENABLED_BY_DEFAULT),
         {"9.9.9.9", "149.112.112.112", "2620:fe::fe", "2620:fe::9"},
         /*dns_over_tls_hostnames=*/{"dns.quad9.net", "dns9.quad9.net"},
         "https://dns.quad9.net/dns-query",
